### ISSUE TITLE
Multiple conditions issue

### ISSUE THREAD

**Issue by maestr0ua** [Raizlabs/DBFlow#653]
I`m trying to make a query something like this:
**SELECT** * **FROM** 'PersonnelTable' **WHERE** country **LIKE** '%UAE%' **AND** region **LIKE** '%Abu-dabi%' **AND** (lastName **LIKE** '%ha%' **OR** firstName **LIKE** '%ha%')

Here is how I`m trying:

```
ConditionGroup conditionGroup = ConditionGroup.clause();
conditionGroup.and(Condition.column(new NameAlias("country")).like("%"+ countryName +"%"));
conditionGroup.and(Condition.column(new NameAlias("region")).like("%"+ regionName +"%"));
conditionGroup.and(Condition.column(new NameAlias("firstName")).like("%"+ firstName +"%"));
conditionGroup.or(Condition.column(new NameAlias("lastName")).like("%"+ lastName +"%"));

SQLite.select().from(PersonnelTable.class)
            .where(conditionGroup)
            .queryList();

```

But the problem is I cannot understand how to put brackets to condition and now my query looks like:
**SELECT** * **FROM** 'PersonnelTable' **WHERE** country **LIKE** '%UAE%' **AND** region **LIKE** '%Abu-dabi%' **AND** lastName **LIKE** '%ha%' OR firstName **LIKE** '%ha%'

without any brackets.

**Comment by gauravsak** [comment]
@maestr0ua Create a separate condition group for the name conditions and pass it as a second parameter in the where clause.

**Comment by agrosner** [comment]
```java
ConditionGroup conditionGroup = ConditionGroup.clause()
      .and(Condition.column(new NameAlias("country")).like("%"+ countryName +"%"))
      .and(Condition.column(new NameAlias("region")).like("%"+ regionName +"%"))
      .and(ConditionGroup.clause().
            .and(Condition.column(new NameAlias("firstName")).like("%"+ firstName +"%"))
           .or(Condition.column(new NameAlias("lastName")).like("%"+ lastName +"%"));

SQLite.select().from(PersonnelTable.class)
            .where(conditionGroup)
            .queryList();

````

**Comment by agrosner** [comment]
Also id strongly recommend you use the generated `Property` fields in the `_Table` classes.

**Comment by maestr0ua** [comment]
Thank you for your answer, but now I`m getting error at query execution.

```
  ConditionGroup conditionGroup = ConditionGroup.clause()
                .and(LocationTable_Table.type.eq(Constants.COUNTRY))
                .and(ConditionGroup.clause()
                        .and(LocationTable_Table._id.eq(ids.get(0)))
                        .or(LocationTable_Table._id.eq(ids.get(1))));

       SQLite.select().from(LocationTable.class)
                .where(conditionGroup)
                .queryList()

```

android.database.sqlite.SQLiteException, cause - null, message - near "AND": syntax error (code 1): , while compiling: SELECT * FROM `LocationTable` WHERE `type`='country' AND `_id`='572b60d432f3d0985041994c' OR `_id`='572b60d432f3d0985041994d' AND

Somehow in the end of ther query shows up operator AND

**Comment by jamesisaac** [comment]
Experienced the same issue in 3.0.0-beta5 (trailing AND when constructing a query like in this example).  Refactoring out the highest level ConditionGroup solves it for me.

```
SQLIte.select().from(...)
    .where(...)
    .and(ConditionGroup.clause()
        .or(...)
        .or(...)
    )
```

**Comment by raphaelbgr** [comment]
ConditionGroup not available in 4.2.4, what`s the equivalent for this?

**Comment by harinugroho** [comment]
it change to OperatorGroup @raphaelbgr [DBFlow 4.0 Migration guide](https://agrosner.gitbooks.io/dbflow/content/Migration4Guide.html)
